src/services/email.service.js

```diff
@@ -0,0 +1,42 @@
+import { config } from 'dotenv';
+import sendgrid from '@sendgrid/mail';
+import Debug from 'debug';
+
+config();
+
+const debug = Debug('dev');
+
+const { SENDGRID_API_KEY, SENDGRID_EMAIL, SENDGRID_ACTIVATION_TEMPLATE } = process.env;
+const templates = { accountActivation: SENDGRID_ACTIVATION_TEMPLATE };
+
+
+/**
+     * Initialises the event .
+     * @param {Object} message - The message payload.
+     * @param {String} type - the type of message which corresponds to the mail template
+     * @param {Object | Array} data - can be a list for bulk sending. must contain email parameter
+     * @return {promise} - Returns a promise that get resolcved to a success object
+*/
+export default async ({ type, data }) => {
+  const template = templates[type];
+
+  if (!template) throw new Error('Email Template not available');
+  if (!data) throw new Error('Mail data cannot be empty');
+  const mailDetails = Array.isArray(data) ? data : [data];
+
+  sendgrid.setApiKey(SENDGRID_API_KEY);
+  const messagePayload = mailDetails.map((singleEmailData) => {
+    const { email } = singleEmailData;
+    return {
+      to: email,
+      from: SENDGRID_EMAIL,
+      templateId: template,
+      dynamic_template_data: {
```

Comment:
I don't understand. Does the Sendgrid API require camel case and snake case in the same payload?

Reply:
yes the dynamic template data is snakecase